geth/jail/internal/loop/loop_test.go

```diff
@@ -0,0 +1,74 @@
+package loop
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+
+	"github.com/status-im/status-go/geth/jail/internal/vm"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestLoopSuite(t *testing.T) {
+	suite.Run(t, new(LoopSuite))
+}
+
+type LoopSuite struct {
+	suite.Suite
+
+	id int64
+
+	loop   *Loop
+	cancel context.CancelFunc
+
+	mockTask     *MockTask
+	loopMockCtrl *gomock.Controller
+}
+
+func (s *LoopSuite) SetupTest() {
+	s.loopMockCtrl = gomock.NewController(s.T())
+	s.mockTask = NewMockTask(s.loopMockCtrl)
+
+	vm := vm.New()
+	s.loop = New(vm)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
+	go s.loop.Run(ctx)
+}
+
+func (s *LoopSuite) TearDownTest() {
+	s.loopMockCtrl.Finish()
+}
+
+func (s *LoopSuite) TestAddAndReady() {
+	s.mockTask.EXPECT().SetID(int64(1)).Times(1)
+	s.mockTask.EXPECT().GetID().Times(1)
+	s.mockTask.EXPECT().Cancel().Times(1)
+
+	err := s.loop.Add(s.mockTask)
+	s.NoError(err)
+
+	err = s.loop.Ready(s.mockTask)
+	s.NoError(err)
+
+	s.cancel()
+
+	// Wait for the context to cancel and loop to close
```

Comment:
Isn't this a left-over from the test splitting?

Reply:
Since `SetupTest` is called before we run each test, a `vm`, `ctx` and `loop` are created for each test and `s.loop.Run(ctx)` is called in a goroutine. Therefore, I think we should close the `loop` at the end of each test to clean up the goroutine.

Also, if we remove it, the function would still have to wait because `loop` will try to process the tasks, but `mock` won't be happy with the extra calls to `task.GetID` and `task.Execute`.

Reply:
Ah, thanks. My question has been less for the `cancel()` but for the sleep without any further check. So it looks a bit like *100 ms should be enough*.